### DISCUSSION TITLE
I may have made some purchases...

### DISCUSSION THREAD

**Discussion by Devasta** [ballsteve/xrust#51] [General]
[http://xml.rs](http://xml.rs)
[http://xslt.rs](http://xslt.rs)

**Comment by ballsteve** [comment]
That's cool!

Now we just a website to go behind them ;-)

Last year I was playing around with Rocket. There are some interesting frameworks being developed for the Web using Rust, so we could look at adopting one of them.

**Reply by Devasta** [reply]
In a perfect world, I would love to have something like XSLTFiddle running with xrust in browser.

http://xsltfiddle.liberty-development.net/

**Reply by ballsteve** [reply]
For a previous project I did a lot of work with Saxon-JS. I don't think XSLT quite works in a browser setting, because you're trying to modify an existing document (the DOM), rather than creating a new document.

I've been working on xrust in the browser, using wasm. I call this 'Action Sheets'. Instead of templates, an action sheet attaches 'behaviours' to nodes in the DOM. These react to DOM events and have a template for how the node (or some other node) is changed.

Now that the networking code for include/import has been separated out from the xrust crate, I can get back on to developing the wasm code.
